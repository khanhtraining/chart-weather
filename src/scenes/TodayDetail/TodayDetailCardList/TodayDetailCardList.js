import React from 'react'
import TodayDetailAdd from './TodayDetailAdd'
import TodayDetailCard from './TodayDetailCard'
import './todaydetailcards.scss'

const TodayDetailCardList = (props) => {
    const { feelsLike, speed } = props
    const itemCardList = [
        {
            title: 'PSI',
            content: feelsLike,
            extra: 'Good',
            contentClassName: 'bg-number'
        },
        {
            title: 'WIND SPEED',
            content: speed,
            extra: 'km/h',
            contentClassName: 'normal-number'
        },
        {
            title: 'DENGUE',
            contentClassName: 'circle-shape'
        }
    ]

    return (
        <React.Fragment>
            {
                <div className='todayDetailCards-wrapper'>
                    {itemCardList.map((item) => {
                        return <TodayDetailCard {...item} />
                    })}
                    <TodayDetailAdd />
                </div>
            }
        </React.Fragment>
    )
}

export default TodayDetailCardList
