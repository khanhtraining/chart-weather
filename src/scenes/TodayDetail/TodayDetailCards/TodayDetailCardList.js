import React from 'react'
import TodayDetailAdd from './TodayDetailAdd'
import TodayDetailCard from './TodayDetailCard'
import './todaydetailcards.scss'

const TodayDetailCardList = (props) => {
    const itemCardList = [
        {
            title: 'PSI',
            content: Math.round((props.todayDeTail?.main?.feels_like) / 10) || 23,
            extra: 'Good',
            contentClassName: 'bg-number'
        },
        {
            title: 'WIND SPEED',
            content: Math.round(props.todayDeTail?.wind?.speed) || '--',
            extra: 'km/h',
            contentClassName: 'normal-number'
        },
        {
            title: 'DENGUE',
            contentClassName: 'circle-shape'
        }
    ]

    return (
        <React.Fragment>
            {
                <div className='todayDetailCards-wrapper'>
                    {itemCardList.map((item) => {
                        return <TodayDetailCard {...item} />
                    })}
                    <TodayDetailAdd />
                </div>
            }
        </React.Fragment>
    )
}

export default TodayDetailCardList
