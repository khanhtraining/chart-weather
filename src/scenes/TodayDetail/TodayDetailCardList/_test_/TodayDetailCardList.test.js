import React from 'react'
import renderer from 'react-test-renderer'
import TodayDetailCardList from '../TodayDetailCardList'

describe('snapshot testing', () => {
    test('snapshot for today detail card_list component', () => {
        const renderedComponent = renderer.create(<TodayDetailCardList/>).toJSON()
        expect(renderedComponent).toMatchSnapshot()
    })
})
