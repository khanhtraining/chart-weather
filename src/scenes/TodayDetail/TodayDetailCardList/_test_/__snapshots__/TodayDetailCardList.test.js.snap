// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for Today Detail Card List component 1`] = `
<div
  className="todayDetailCards-wrapper"
>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      PSI
    </div>
    <div
      className="content bg-number"
    >
      NaN
    </div>
    <div
      className="extra"
    >
      Good
    </div>
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      WIND SPEED
    </div>
    <div
      className="content normal-number"
    >
      NaN
    </div>
    <div
      className="extra"
    >
      km/h
    </div>
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      DENGUE
    </div>
    <div
      className="content circle-shape"
    />
    <div
      className="extra"
    />
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="add-icon"
    />
    <div
      className="add-text"
    >
      Add
    </div>
  </div>
</div>
`;

exports[`snapshot testing snapshot for Today Detail component 1`] = `
<div
  className="todayDetailCards-wrapper"
>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      PSI
    </div>
    <div
      className="content bg-number"
    >
      NaN
    </div>
    <div
      className="extra"
    >
      Good
    </div>
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      WIND SPEED
    </div>
    <div
      className="content normal-number"
    >
      NaN
    </div>
    <div
      className="extra"
    >
      km/h
    </div>
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      DENGUE
    </div>
    <div
      className="content circle-shape"
    />
    <div
      className="extra"
    />
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="add-icon"
    />
    <div
      className="add-text"
    >
      Add
    </div>
  </div>
</div>
`;

exports[`snapshot testing snapshot for today detail card_list component 1`] = `
<div
  className="todayDetailCards-wrapper"
>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      PSI
    </div>
    <div
      className="content bg-number"
    >
      NaN
    </div>
    <div
      className="extra"
    >
      Good
    </div>
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      WIND SPEED
    </div>
    <div
      className="content normal-number"
    >
      NaN
    </div>
    <div
      className="extra"
    >
      km/h
    </div>
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="title"
    >
      DENGUE
    </div>
    <div
      className="content circle-shape"
    />
    <div
      className="extra"
    />
  </div>
  <div
    className="todayDetailCards-wrapper-item"
  >
    <div
      className="add-icon"
    />
    <div
      className="add-text"
    >
      Add
    </div>
  </div>
</div>
`;
